test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/fields/Context.java

```diff
@@ -46,27 +52,41 @@ public DataSourceResponse.ChildFieldGenerator childFieldGenerator() {
         return childFieldGenerator;
     }
 
-    public DataSourceResponse.FieldTypeGenerator fieldTypeGenerator() {
-        return fieldTypeGenerator;
+    public DataSourceResponse.FieldTypeGenerator fieldTypeGenerator(DynamicMapping dynamicMapping) {
+        return specification.dataSource().get(new DataSourceRequest.FieldTypeGenerator(dynamicMapping));
     }
 
-    public Context subObject() {
-        return new Context(specification, objectDepth + 1, nestedFieldsCount);
+    public Context subObject(DynamicMapping dynamicMapping) {
+        return new Context(specification, objectDepth + 1, nestedFieldsCount, dynamicMapping);
     }
 
-    public Context nestedObject() {
+    public Context nestedObject(DynamicMapping dynamicMapping) {
         nestedFieldsCount.incrementAndGet();
-        return new Context(specification, objectDepth + 1, nestedFieldsCount);
+        return new Context(specification, objectDepth + 1, nestedFieldsCount, dynamicMapping);
+    }
+
+    public boolean shouldAddDynamicObjectField(DynamicMapping dynamicMapping) {
+        if (objectDepth >= specification.maxObjectDepth() || dynamicMapping == DynamicMapping.FORBIDDEN) {
```

Comment:
What value we use for max depth? Shall this be in the 5-10 range?

Reply:
We use 2 because when you increase it, it quickly starts producing gigantic objects.